manifests/instances/accounts.pp

```diff
@@ -0,0 +1,128 @@
+#
+# @param instance_name
+#   The unique name of the instance
+#
+# @param base_dn
+#   The 'base' DN component of the directory server
+#
+# @param root_dn
+#   The default administrator DN for the directory server
+#
+#   * NOTE: To work around certain application bugs, items with spaces may not
+#     be used in this field.
+#
+# @param root_dn_password
+#   The password for the the ``$root_dn``
+#
+#   * NOTE: To work around certain application bugs, items with spaces may not
+#     be used in this field.
+#
+# @param listen_address
+#   The IP address upon which to listen
+#
+# @param bootstrap_ldif_content
+#   The content that should be used to initialize the directory
+#
+# @param enable_tls
+#   Wether to configure the server to use TLS and also how to copy the
+#   pki certificates.
+#     simp => Will enable TLS and copy the certificates out from the
+#             puppetserver.
+#     true => Will enable TLS and copy the certificates from a local
+#             directory on the server.
+#     false => Will not enable TLS
+#
+# @param port
+#   The port upon which to accept normal/STARTTLS connections
+#
+# @param secure_port
+#   The port upon which to accept LDAPS connections
+#
+# @param password_policy
+#  Settings for the password policy.  The defaults in the module data
+#  are set to meet most compliance standards.
+#
+# @param instance_params
+#   Any other arguments that you wish to pass through directly to the
+#   `ds389::instance` Defined Type.
+#
+# @param users_group_id
+#   The group ID of the "users" group created in the install.
+#
+# @param administrators_group_id
+#   The group ID of the "administrators" group created in the install.
+#   The pupmod-simp-simp module configures permissions on systems using
+#   simp with the admin.pp manifest.
+#
+# @param firewall
+#   Wether or not to configure access through the firewall.
+#
+# @param trusted_nets
+#   Which networks to all access through the firewall.
+#
+# @author https://github.com/simp/pupmod-simp-ds389/graphs/contributors
+#
+class simp_ds389::instances::accounts (
+  String[1]                      $instance_name           = 'accounts',
+  String[2]                      $base_dn                 = simplib::lookup('simp_options::ldap::base_dn', { 'default_value' => sprintf(simplib::ldap::domain_to_dn($facts['domain'], true)) }),
+  String[2]                      $root_dn                 = 'cn=Directory_Manager',
+  String[2]                      $root_pw                 = simplib::passgen('simp_ds389-rootdn_accounts', { 'length' => 64, 'complexity' => 0 }),
+  String[2]                      $bind_dn                 = simplib::lookup('simp_options::ldap::bind_dn', { 'default_value'   => "cn=hostAuth,ou=Hosts,${base_dn}" }),
+  String[1]                      $bind_pw                 = simplib::lookup('simp_options::ldap::bind_hash', { 'default_value' => simplib::passgen("ds389_${instance_name}_bindpw", {'length' => 64})}),
```

Comment:
This is confusing.  Why is it looking up `simp_options::ldap::bind_hash` instead of `simp_options::ldap::bind_pw`?  Is this a typo or is the parameter missnamed?

Reply:
This is a typo